.github/workflows/release_github.yaml

```diff
@@ -11,23 +10,56 @@ on:
       - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
 
 jobs:
-  code-checks:
-    uses: ./.github/workflows/ci-checks.yaml
 
-  release-checks:
+  tests:
+    uses: ./.github/workflows/ci-tests.yaml
+
+  release_checks:
     name: Checks before pkg build
     runs-on: ubuntu-latest
     timeout-minutes: 5
-    needs: code-checks
+    strategy:
+      fail-fast: true
     steps:
-      - id: version-vs-tag-check
+      - id: checkout
+        name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ env.BRANCH }}
+          fetch-depth: 0
+      - id: setup_python
+        name: Setup Python
+        uses: actions/setup-python@v2.2.2
+        with:
+          python-version: '3.10'
+          architecture: x64
+      - id: install_environment
+        name: Set up build environment
+        run: |
+          make setup
+
+      - id: package_metadata
+        name: Fetch package metadata
+        run: |
+          NAME=$(poetry version | awk '{print $1}')
+          VERSION=$(poetry version | awk '{print $2}')
+          echo "name=$NAME" >> "$GITHUB_OUTPUT"
+          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+
+      - id: version_vs_tag_check
         name: Check if tag version matches project version
         run: |
+          VERSION=${{ steps.package_metadata.outputs.version }}
+          BRANCH=${{ env.BRANCH }}
           if [[ "$BRANCH" != "v$VERSION" ]]; then exit 1; fi
 
+    outputs:
+      package_name: ${{ steps.package_metadata.outputs.name }}
+      package_version: ${{ steps.package_metadata.outputs.version }}
+
   autorelease:
     name: Release the package on github
-    needs: release-checks
+    needs: [release_checks, tests]
     runs-on: ubuntu-latest
```

Comment:
maybe take a pinned major version, I'm not sure what `ubuntu-latest` might be. Does it mean interim releases or LTS?